projects/storefrontlib/src/lib/cms/services/cms-routes.service.spec.ts

```diff
@@ -0,0 +1,121 @@
+import { TestBed } from '@angular/core/testing';
+
+import { CmsRoutesService } from './cms-routes.service';
+import { Router } from '@angular/router';
+import { CmsRoute, CmsService, PageType } from '@spartacus/core';
+import { CmsMappingService, PageLayoutComponent } from '@spartacus/storefront';
+import { of } from 'rxjs';
+import createSpy = jasmine.createSpy;
+
+describe('CmsRoutesService', () => {
+  let service: CmsRoutesService;
+  let mockRouter;
+
+  const mockPageContext = {
+    type: PageType.CONTENT_PAGE,
+    id: '/testRoute2'
+  };
+
+  const mockRouterConfig: CmsRoute[] = [
+    { path: 'route1' },
+    {
+      path: 'testRoute',
+      data: {
+        cxCmsContext: { type: PageType.CONTENT_PAGE, id: '/testRoute' }
+      }
+    }
+  ];
+
+  const mockCmsMapping = {
+    getRoutesFromPageData: () => [{ path: 'route1' }]
```

Comment:
I think that `route1` is a bit unfortunate here, because in line 20 we have an ordinary `route1` path.

Reply:
Thanks, changed to 'sub-route'